code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -93,13 +93,14 @@
 	if(resting) //Resting doubles sunder recovery
 		sunder_recov *= 2
 
-	if(locate(/obj/effect/alien/weeds) in loc) //Weeds double sunder recovery
+	var/on_weeds = locate(/obj/effect/alien/weeds) in loc
+	if(on_weeds) //Weeds double sunder recovery
 		sunder_recov *= 2
 
 	if(recovery_aura)
 		sunder_recov *= 1 + recovery_aura * 0.1 //10% bonus per rank of recovery aura
 
-	SEND_SIGNAL(src, COMSIG_XENOMORPH_SUNDER_REGEN, src)
+	SEND_SIGNAL(src, COMSIG_XENOMORPH_SUNDER_REGEN, src, on_weeds)
```

Comment:
why

Reply:
you don't need this param, you also dont need to send src as a param.